### PR TITLE
gh-112240: Add option to calendar module CLI to specify the weekday to start each week

### PR DESCRIPTION
**Pull request by planet36**
This PR adds `--first-weekday` and `-f` options to the calendar module CLI.  These take an integer representing the weekday to start each week.

* Issue: gh-112240

----
:books: Documentation preview :books:: https://cpython-previews--112241.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/calendar.rst [hunk 1]
```diff
@@ -586,10 +586,16 @@ The following options are accepted:
    or as an HTML document.
 
 
+.. option:: --firstweekday WEEKDAY, -f WEEKDAY
+
+   The weekday to start each week.
+   Must be a number between 0 and 6.
+   Defaults to 0.
```
Comment (serhiy-storchaka) [inline Doc/library/calendar.rst]: Please clarify that 0 is Monday and 6 is Sunday. Also add this in the option help.

File: Doc/library/calendar.rst [hunk 2]
```diff
@@ -586,10 +586,16 @@ The following options are accepted:
    or as an HTML document.
 
 
+.. option:: --firstweekday WEEKDAY, -f WEEKDAY
```
Comment (hugovk) [inline Doc/library/calendar.rst]: Let's hyphenate this for readability and accessibility (for example, it helps people using screen readers):
```suggestion
.. option:: --first-weekday WEEKDAY, -f WEEKDAY
```

The GNU reference manual also recommends it:

> GNU adds _long options_ to these conventions. Long options consist of `--` followed by a name made of alphanumeric characters and dashes. Option names are typically one to three words long, with hyphens to separate words. Users can abbreviate the option names as long as the abbreviations are unique.


https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

File: Lib/calendar.py [hunk 3]
```diff
@@ -734,10 +734,15 @@ def main(args=None):
         choices=("text", "html"),
         help="output type (text or html)"
     )
+    parser.add_argument(
+        "-f", "--firstweekday",
```
Comment (hugovk) [inline Lib/calendar.py]:
```suggestion
        "-f", "--first-weekday",
```

File: Lib/calendar.py [hunk 4]
```diff
@@ -775,6 +781,7 @@ def main(args=None):
             cal = LocaleTextCalendar(locale=locale)
         else:
             cal = TextCalendar()
+        cal.setfirstweekday(options.firstweekday)
```
Comment (hugovk) [inline Lib/calendar.py]:
```suggestion
        cal.setfirstweekday(options.first_weekday)
```

File: Misc/NEWS.d/next/Library/2023-11-18-16-30-21.gh-issue-112240.YXS0tj.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Add option to calendar module CLI to specify the weekday to start each week
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2023-11-18-16-30-21.gh-issue-112240.YXS0tj.rst]:
```suggestion
Add option to calendar module CLI to specify the weekday to start each week.
```

(Feel free to also append "Patch by Steven Ward.")

File: Lib/calendar.py [hunk 6]
```diff
@@ -761,6 +766,7 @@ def main(args=None):
             cal = LocaleHTMLCalendar(locale=locale)
         else:
             cal = HTMLCalendar()
+        cal.setfirstweekday(options.firstweekday)
```
Comment (hugovk) [inline Lib/calendar.py]:
```suggestion
        cal.setfirstweekday(options.first_weekday)
```

File: Doc/library/calendar.rst (hunk 1 above)
Comment (planet36) [reply Doc/library/calendar.rst]: I made changes based on your feedback.

File: Doc/library/calendar.rst (hunk 2 above)
Comment (planet36) [reply Doc/library/calendar.rst]: I applied your changes.

File: Lib/calendar.py (hunk 3 above)
Comment (planet36) [reply Lib/calendar.py]: I applied your changes.

File: Lib/calendar.py (hunk 4 above)
Comment (planet36) [reply Lib/calendar.py]: I applied your changes.

File: Misc/NEWS.d/next/Library/2023-11-18-16-30-21.gh-issue-112240.YXS0tj.rst (hunk 5 above)
Comment (planet36) [reply Misc/NEWS.d/next/Library/2023-11-18-16-30-21.gh-issue-112240.YXS0tj.rst]: I applied your changes.

File: Lib/calendar.py (hunk 6 above)
Comment (planet36) [reply Lib/calendar.py]: I applied your changes.

File: Doc/library/calendar.rst [hunk 7]
```diff
@@ -586,10 +586,16 @@ The following options are accepted:
    or as an HTML document.
 
 
+.. option:: --first-weekday WEEKDAY, -f WEEKDAY
+
+   The weekday (``0`` is Monday, ``6`` is Sunday) to start each week.
+   Must be a number between 0 and 6.
```
Comment (serhiy-storchaka) [inline Doc/library/calendar.rst]: What do you think about such variant?
```suggestion
   The weekday to start each week.
   Must be a number between 0 (Monday) and 6 (Sunday).
```

File: Lib/calendar.py [hunk 8]
```diff
@@ -734,10 +734,15 @@ def main(args=None):
...
+    parser.add_argument(
+        "-f", "--first-weekday",
+        type=int, default=0,
+        help="weekday (0 is Monday, 6 is Sunday) to start each week (default 0)"
+    )
     parser.add_argument(
         "year",
         nargs='?', type=int,
-        help="year number (1-9999)"
+        help="year number"
```
Comment (serhiy-storchaka) [inline Lib/calendar.py]: Why is it removed?

File: Lib/calendar.py [hunk 9]
```diff
@@ -193,7 +193,7 @@ class Calendar(object):
     """
 
     def __init__(self, firstweekday=0):
-        self.firstweekday = firstweekday # 0 = Monday, 6 = Sunday
+        self.firstweekday = firstweekday
```
Comment (serhiy-storchaka) [inline Lib/calendar.py]: Why is it removed?

File: Lib/calendar.py (hunk 8 above)
Comment (planet36) [reply Lib/calendar.py]: The limit was removed in 66c88ce30ca.

File: Lib/calendar.py (hunk 9 above)
Comment (planet36) [reply Lib/calendar.py]: It seemed redundant.  I'll put it back if you wish.

File: Lib/calendar.py (hunk 8 above)
Comment (serhiy-storchaka) [reply Lib/calendar.py]: Good point.

This PR will only be merged in the main branch. Do you mind to create a separate PR that fixes this help for backporting?

File: Lib/calendar.py (hunk 9 above)
Comment (serhiy-storchaka) [reply Lib/calendar.py]: It is unrelated to this PR.

Comment (planet36) [reply Lib/calendar.py]: I restored the comment.

File: Lib/calendar.py (hunk 8 above)
Comment (planet36) [reply Lib/calendar.py]: I _could_ make a separate PR.  How is the process different from a normal PR that goes into main?  This is my first PR for python.

Comment (serhiy-storchaka) [reply Lib/calendar.py]: The simplest way -- you only need to create a trivial PR for main. I'll merge it and create backports, then I'll merge main into this branch and merge it into main.

Comment (planet36) [reply Lib/calendar.py]: I made this https://github.com/python/cpython/pull/114719

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @planet36. And congratulate you with your first commit in CPython. Well, technically not first, but actually first.

**Pull request merged**